locallib.php

```diff
@@ -237,9 +237,9 @@ function bigbluebuttonbn_get_recordings_array($meetingids, $recordingids = []) {
  */
 function bigbluebuttonbn_get_recordings_array_fetch($meetingidsarray) {
     if ((defined('PHPUNIT_TEST') && PHPUNIT_TEST)
-            || defined('BEHAT_SITE_RUNNING')
```

Comment:
I missed this segment of code in the other PR.

Is there a better way of faking the response other than hard-coding the behaviour? I don't think the code for testing should be tied to the code used for the actual functionality. Not familiar on how things are done here, but on any other framework the Unit Tests are entirely independent.

Reply:
Just a warning: the actual code here was created in another ticket. I just pushed a new version which does not have the unwanted space changes, so the changeset here is smaller.

You are right in saying that this is not an ideal situation but I am not sure about the best approach ref. this issue.
I wanted to be able to mock the return value of this function in both php unit and behat tests so we could then create fake recordings.

I suppose looking back at it that we should also refactor the couple of functions that make API and regroup them in a class/singleton interfacing with Bigbluebutton. This class can  then be overridden during test (the base method such as bigbluebuttonbn_wrap_xml_load_file but also higher level method such as bigbluebuttonbn_get_recordings_array_fetch or any which are using the bigbluebuttonbn_wrap_xml_load_file function).
This would probably help with mocking (phpunit), not sure yet about the approach with behat (for which we need also to have a preset of recordings for the new tests in this ticket).